projects/knora-ui/src/lib/viewer/values/color-value/color-picker/color-picker.component.ts

```diff
@@ -121,17 +109,17 @@ export class ColorPickerComponent extends _MatInputMixinBase implements OnInit,
   }
 
   @Input()
-  get value(): ColorPicker | null {
+  get value(): string | null {
     const colorValue = this.colorForm.value;
     if (colorValue !== null) {
-      return new ColorPicker(colorValue.color);
+      return colorValue;
     }
     return null;
   }
 
-  set value(colorValue: ColorPicker | null) {
+  set value(colorValue: string | null) {
     if (colorValue !== null) {
-      this.colorForm.setValue({ color: colorValue.color });
+      this.colorForm.setValue({ color: colorValue });
```

Comment:
Just make the value a string, not an object: `this.colorForm.setValue(colorValue);`

Reply:
otherwise the getter returns an object and this is why you get a 

> Fatal error in JsonConvert. Failed to map the JavaScript instance of class "UpdateColorValue" to JSON because of a type error.

	Class property: 
		color

	Class property value: 
		[object Object]

	Expected type: 
		string

	Runtime type: 
		object

	JSON property: 
		http://api.knora.org/ontology/knora-api/v2#colorValueAsColor

	Reason: Given object does not match the expected primitive type.

Reply:
in case you need an object because of the color picker, then change the getter so it returns only the string value of the object.

Reply:
I wanted to get rid of the object but the signature of setValue() expect it:
<img width="1041" alt="Screenshot 2020-03-11 at 14 57 23" src="https://user-images.githubusercontent.com/29532462/76424550-ac9f5480-63a8-11ea-9aec-4d429f3b12dd.png">

Reply:
it0s fine if you set it as an object, as long as the setter expects a string.

Reply:
I think it's fine as it is now.